src/Maestro/Types/V1/Blocks.hs

```diff
@@ -0,0 +1,60 @@
+-- | Module to define types for /\"Blocks\"/ category endpoints defined at [docs.gomaestro.org](https://docs.gomaestro.org/docs/category/blocks).
+
+module Maestro.Types.V1.Blocks
+  ( BlockDetails (..)
+  , TimestampedBlockDetails (..)
+  ) where
+
+import           Data.Text               (Text)
+import           Data.Word               (Word64, Word32)
+import           Deriving.Aeson
+import           Maestro.Types.Common
+import           Maestro.Types.V1.Pools   (PoolId)
+import           Maestro.Types.V1.Common
+
+-- | Complete block Details when queried by its hash or height.
+data BlockDetails = BlockDetails
+  { _blockDetailsHash :: !BlockHash
+  -- ^ Block hash (identifier)
+  , _blockDetailsHeight :: !BlockHeight
+  -- ^ Block height
+  , _blockDetailsAbsoluteSlot :: !Word64
+  -- ^ Absolute slot of the block which includes the transaction
+  , _blockDetailsBlockProducer :: !(Maybe (Bech32StringOf PoolId))
+  -- ^ Block producer identifier name
+  , _blockDetailsConfirmations :: !Word64
+  -- ^ Total number of confirmations
+  , _blockDetailsEpoch :: !EpochNo
+    -- ^ Epoch number
+  , _blockDetailsEpochSlot :: !EpochSize
```

Comment:
`SlotNo`?

Reply:
Mmm, right, thanks for these suggestions. I wasn't sure about which types. For instance, for this one I followed the "example" https://github.com/maestro-org/haskell-sdk/blob/deb1e133cd7de989e9e85c2b44e9361f52e93776/src/Maestro/Types/V0/Pool.hs#L56

It is V0, but maybe it's something to review.

Reply:
`SlotNo` I believe is appropriate. `EpochSize` IIRC was to denote length of an epoch.